distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/util/ServerStateUtil.java

```diff
@@ -52,13 +57,44 @@ public static ServerState waitForServerState(int expected, int attempts, DummyCl
 			if (s != null) {
 				if (expected == NO_STATE
 						|| expected == s.getState()) {
-					client.clearState();
 					return s;
 				}
 			}
-            System.out.println("Server state: " + ServerStateUtil.toStateString(s));
             Thread.sleep(WAIT_FOR_SERVERSTATE);
         }
         throw new AssertionError("Waiting for server state to change to " + expected + " timed out");
     }
+    
+    public static ServerState waitForDeployables(int attempts, DummyClient client) throws InterruptedException {
+        int tries = attempts;
+        
+        while(tries > 0) {
+            tries--;
+            ServerState s = client.getStateObject();
+			if (s != null) {
+				if (s.getDeployableStates().size() > 0) {
+					return s;
+				}
+			}
+            Thread.sleep(1000);
+        }
+        throw new AssertionError("There are no deployables for the server"); 
+    }
+    
+    public static ServerState waitForDeployablePublishState(int expected, int attempts, DummyClient client) throws Exception {
```

Comment:
This looks like ```waitForDeployables``` with an extra condition. If there was a value that would indicate ```expected``` as any possible value, it could serve as both.

Reply:
nice solution, I tried to follow the hints.. check it out.